File: Data Structures & Algorithms/binary-search/submission-0.java
class Solution {
    public int search(int[] nums, int target) {
        int i = 0;
        int j = nums.length-1;

        while(i<=j){
            int m = i + (j-i)/2;
            if(target < nums[m]){
                j = m-1;
            }else if(target > nums[m]){
                i = m+1;
            }else{
                return m;
            }
        }
        return -1;
    }
}

File: Data Structures & Algorithms/combination-target-sum/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @returns {number[][]}
     */
    combinationSum(nums, target) {
         let res = [];
         let curr = [];
         this.backtrack(nums, target, res, curr, 0);
         return res;

    }
    
    backtrack(nums, target, res, curr, index){
        if(target === 0){
            res.push([...curr])
        }else if(target < 0 || index >= nums.length){
            return
        }else{
            curr.push(nums[index])
            this.backtrack(nums, target - nums[index], res, curr, index);
            curr.pop();
            this.backtrack(nums, target, res, curr, index+1)
        }
    }
}
